app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java

```diff
@@ -248,17 +251,59 @@ private void processAppListData(@NonNull Response<ResponseBody> response) {
                         }
                     }
                 }
-                Timber.i("Synchronization complete!");
 
-                        // Update time of last synchronization
-                        AppPrefs.saveLastSyncTime(Calendar.getInstance().getTimeInMillis());
-                    }
-                } catch (JSONException e) {
-                    Timber.e(e);
-                } catch (IOException e) {
-                    Timber.e(e);
+                if (appCollectionId > 0) {
+                    // Custom Project AppCollection
+
+                    // Store JSON response in a file on the /sdcard so that the Custom Launcher can fetch the AppCollection
+                    // TODO: create Android library for providing the AppCollection instead of writing/reading file
+                    appCollectionService.getAppCollection(appCollectionId, AppPrefs.getLicenseEmail(), AppPrefs.getLicenseNumber())
+                            .enqueue(new Callback<ResponseBody>() {
+                                @Override
+                                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                                    Timber.i("appCollectionService.getAppCollection onResponse");
+
+                                    ResponseBody responseBody = response.body();
+                                    try {
+                                        String jsonReponse = responseBody.string();
+                                        Timber.i("jsonReponse: " + jsonReponse);
+
+                                        File fileDir = new File(Environment.getExternalStorageDirectory() + "/.elimu-ai/appstore/");
+                                        Timber.i("fileDir: " + fileDir);
+                                        if (!fileDir.exists()) {
+                                            fileDir.mkdirs();
+                                        }
+
+                                        File jsonFile = new File(fileDir, "app-collection.json");
+                                        Timber.i("jsonFile: " + jsonFile);
+                                        try {
+                                            FileUtils.writeStringToFile(jsonFile, jsonReponse, "UTF-8", true);
```

Comment:
You are appending json string to the file.
What if the app collection json is saved for the first time, then next time you launch the app, the collection content is updated?
The new json string will be appended to the existing file and it gives a wrong json.

Reply:
Ah, good observation, thanks! This has now been fixed.